exercises/Anthony_Lim/ex02_oo_basics/src/shape_sorter.cpp

```diff
@@ -0,0 +1,85 @@
+//This is the introduction exercise 2. The work has been completed by Anthony Lim 
+/*
+------------------------------------------------------------------------------
+Include statements:
+
+ 	internal headers - same project
+ 	internal headers - different project
+ 	3rd part headers
+	system
+-------------------------------------------------------------------------------
+*/
+
+#include "shape_sorter.h" 
+#include<string> 
+#include<algorithm> 
+/*
+Aurthor: Anthony Lim
+data: 10/01/2017
+*/
+
+/*
+This body file is for the shape_sorter class 
+*/
+void ShapeSorter::PrintMatchedShape(std::string this_shape) const 
+{
+	for(int j =0;j<m_shapes.size();j++)
+	{
+		if(this_shape == m_shapes[j]->GetName())
+		{
+			m_shapes[j]->ReportShape();
+		}
+	}	
+}
+void  ShapeSorter::PrintMatchedSides(int sides) const
+{	
+	for(const auto tmpShape:m_shapes )
+	{
+		if(sides == tmpShape->GetNumSides())
+		{
+			tmpShape->ReportShape();
+		}
+	}	
+}
+
+// this method and the method below could be merged, however I did not know how to pass the method (area or perimeter) to the ordering method. 
+// could have used a map to auto-order the data. However, some data would be lost. 
+void ShapeSorter::PrintOrderByArea()
+{	//order results
+	for(auto k:m_shapes ) // using std::Sort causes problems with the comparison as we wnat to move the object but just compare one particular method => done it explicitly. 
```

Comment:
`k` does not seem to be used. I haven't tested it yet, but does your method work as expected?